src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/WindowsRuntime/Generated/WinRT.cs

```diff
@@ -0,0 +1,518 @@
+            _fileName = fileName;
+
+            // Explicitly look for module in the same directory as this one, and
+            // use altered search path to ensure any dependencies in the same directory are found.
+            _moduleHandle = Platform.LoadLibraryExW(System.IO.Path.Combine(_currentModuleDirectory, fileName), IntPtr.Zero, /* LOAD_WITH_ALTERED_SEARCH_PATH */ 8);
+#if !NETSTANDARD2_0 && !NETCOREAPP2_0
+            if (_moduleHandle == IntPtr.Zero)
+            {
+                try 
+	            {	        
+                    // Allow runtime to find module in RID-specific relative subfolder
+                    _moduleHandle = NativeLibrary.Load(fileName, Assembly.GetExecutingAssembly(), null);
+	            }
+                catch (Exception) { }
+            }
+#endif
+            if (_moduleHandle == IntPtr.Zero)
+            {
+                Marshal.ThrowExceptionForHR(Marshal.GetHRForLastWin32Error());
+            }
+
+            _GetActivationFactory = Platform.GetProcAddress<DllGetActivationFactory>(_moduleHandle);
+
+            var canUnloadNow = Platform.GetProcAddress(_moduleHandle, "DllCanUnloadNow");
+            if (canUnloadNow != IntPtr.Zero)
+            {
+                _CanUnloadNow = Marshal.GetDelegateForFunctionPointer<DllCanUnloadNow>(canUnloadNow);
+            }
+        }
+
+        public unsafe (ObjectReference<IActivationFactoryVftbl> obj, int hr) GetActivationFactory(string runtimeClassId)
+        {
+            IntPtr instancePtr;
+            var hstrRuntimeClassId = MarshalString.CreateMarshaler(runtimeClassId);
+            int hr = _GetActivationFactory(MarshalString.GetAbi(hstrRuntimeClassId), out instancePtr);
+            return (hr == 0 ? ObjectReference<IActivationFactoryVftbl>.Attach(ref instancePtr) : null, hr);
+        }
+
+        ~DllModule()
+        {
+            System.Diagnostics.Debug.Assert(_CanUnloadNow == null || _CanUnloadNow() == 0); // S_OK
+            lock (_cache)
+            {
+                _cache.Remove(_fileName);
+            }
+            if ((_moduleHandle != IntPtr.Zero) && !Platform.FreeLibrary(_moduleHandle))
+            {
+                Marshal.ThrowExceptionForHR(Marshal.GetHRForLastWin32Error());
+            }
+        }
+    }
+
+    internal class WeakLazy<T> where T : class, new()
+    {
+        WeakReference<T> _instance = new WeakReference<T>(null);
+        public T Value
+        {
+            get
+            {
+                lock (_instance)
+                {
+                    T value;
+                    if (!_instance.TryGetTarget(out value))
+                    {
+                        value = new T();
+                        _instance.SetTarget(value);
+                    }
+                    return value;
+                }
+            }
+        }
+    }
+
+    internal class WinrtModule
+    {
+        readonly IntPtr _mtaCookie;
+        static Lazy<WinrtModule> _instance = new Lazy<WinrtModule>();
```

Comment:
Maybe this field can be set to readonly

Reply:
This is part of the generated code. I'd prefer to limit stylistic changes while we get everything else in.